### DISCUSSION TITLE
Adding Pagination during search

### DISCUSSION THREAD

**Discussion by ravi-prakash-v** [beckn/protocol-specifications#127] [Core Specification]
Description required

**Comment by ravi-prakash-v** [comment]
@venkatramanm @thejeshgn do you think search api should support pagination? If so, any recommendations?

**Reply by Vedsaga** [reply]
Are we referring at Interface level or at BPP level` to reduce the fetch time and amount of data to fetched`?
This could help to save alot infrastructure cost however one issue could be that, how will decide the relevance of the search query will that be based on Nearest Location or Best Seller or Highest-rated Product or Recently Launched Product or Highest Price or Lowest Price ?  Does BAP have  to provide certain filter before calling search api

**Comment by venkatramanm** [comment]
How many times do we move to next page in a Google search or rather  change our search criteria, I think following this analogy, bap can ask user to filter better. 

However pagination,  seems to be a use case driven feature. If baps find that pagination will be useful, lets invest time to change the spec. 

Having all BPP support it may be difficult. So we may have the spec be able to dictate, what to do when bpp cannot support pagination

May be  max records may be a good thing to do even if paging is not supported. So that the messages are not too big. 

My take is, only if there is a need for paging at a bap, then change the spec.  In any case seems.like nice to have and not must have.
